firestore/transactions_and_batched_writes.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+
+# [START fs_run_simple_transaction]
+def run_simple_transaction project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  city_ref = firestore.doc "cities/SF"
+  firestore.transaction do |tx|
+    new_population = tx.get(city_ref).data[:population] + 1
+    puts "New population is #{new_population}."
+    tx.update(city_ref, { population: new_population})
+  end
+  puts "Ran a simple transaction to update the population field in the SF document in the cities collection."
+end
+# [END fs_run_simple_transaction]
+
+# [START fs_return_info_transaction]
+def return_info_transaction project_id:
+  # project_id = "Your Google Cloud Project ID"
+  updated = nil
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  city_ref = firestore.doc "cities/SF"
+  firestore.transaction do |tx|
```

Comment:
you can add

updated = firestore.transaction do |tx|

and in lines 41/43 change it to

true

false